CiscoCloud/mantl
# elk addon refactor

- [x] Installs cleanly on a fresh build of most recent master branch
- [x] Upgrades cleanly from the most recent release *
- [x] Updates documentation relevant to the changes

- kibana runs via mesos framework (by default)
- elasticsearch framework 1.0.1 (in docker mode)
- elasticsearch client node (load balancer) for improved resilience
- uninstall playbook

Fixes #1208
Fixes #1207
Fixes #1205
Fixes #1201
Fixes #1198
Fixes #1197
Fixes #830

This updates the ELK addon. See `roles/elk/readme.rst' for updated documentation.

*Re-running the ELK addon playbook on a Mantl cluster running the 1.0.3 ELK addon is not supported without manual intervention. See the documentation for the details

Testing:

* Review documentation
* Install the addon (it can take 5+ minutes for the components to become healthy)
* Verify Elasticsearch framework and Kibana UIs are available in Mantl UI
* Verify that the following services are registered and healthy in consul:
  * elasticsearch-client-mantl
  * elasticsearch-executor-mantl
  * elasticsearch-mantl
  * kibana-mantl
  * kibana-mantl-task
* Verify that the following apps are running and healthy in Marathon (in mantl folder):
  * elasticsearch
  * elasticsearch-client
  * kibana
* Verify Kibana UI
  * Verify that it does not prompt you to create an index pattern
  * Verify that the default dashboard was created
  * Verify that it is receiving data from all nodes
* Verify Elasticsearch framework UI
  * Verify that data size and # of documents are increasing
* Uninstall addon (see docs)
  * Verify all components above are removed
